lib/iris/cube.py

```diff
@@ -3131,6 +3084,10 @@ def __eq__(self, other):
         if isinstance(other, Cube):
             result = self.metadata == other.metadata
 
+            # check the cube fill-value.
+            if result:
+                result = self.fill_value == other.fill_value
```

Comment:
Thanks @pp-mo! I think pretty much all of this particular discussion is now diverging away from what's key to this PR (we can [deal with it here](https://github.com/SciTools/iris/issues/2497) instead).

My preference then is that this PR _does not change current cube equality behaviour_.